lib/src/neu/parser.ts

```diff
@@ -0,0 +1,68 @@
+import * as path from "path";
+import { CompilerOptions, Project, ts } from "ts-morph";
+import { Contract } from "./definitions";
+import { parseContract } from "./parsers/contract-parser";
+
+export function parse(sourcePath: string): Contract {
+  const project = createProject();
+
+  // Add all dependent files that the project requires
+  const sourceFile = project.addExistingSourceFile(sourcePath);
+  project.resolveSourceFileDependencies();
+
+  // Validate that the project has no TypeScript syntax errors
+  validateProject(project);
+
+  const { contract, lociTable } = parseContract(sourceFile);
+
+  return contract;
+}
+
+/**
+ * Create a new project configured for Spot
+ */
+function createProject(): Project {
+  const compilerOptions: CompilerOptions = {
+    target: ts.ScriptTarget.ESNext,
+    module: ts.ModuleKind.CommonJS,
+    strict: true,
+    noImplicitAny: true,
+    strictNullChecks: true,
+    strictFunctionTypes: true,
+    strictPropertyInitialization: true,
+    noImplicitThis: true,
+    alwaysStrict: true,
+    noImplicitReturns: true,
+    noFallthroughCasesInSwitch: true,
+    moduleResolution: ts.ModuleResolutionKind.NodeJs,
+    experimentalDecorators: true,
+    baseUrl: "./",
+    paths: {
+      "@airtasker/spot": [path.join(__dirname, "../lib")]
+    }
+  };
+
+  // Creates a new typescript program in memory
+  return new Project({ compilerOptions });
```

Comment:
Why does it need a new typescript program? I might need to be shown what this means.

Reply:
As discussed offline, this holds the typescript AST of the contract files in memory